packages/boot/src/booters/interceptor.booter.ts

```diff
@@ -0,0 +1,78 @@
+// Copyright IBM Corp. 2018. All Rights Reserved.
+// Node module: @loopback/boot
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {
+  BindingScope,
+  Constructor,
+  createBindingFromClass,
+  GLOBAL_INTERCEPTOR_NAMESPACE,
+  inject,
+  Interceptor,
+  Provider,
+} from '@loopback/context';
+import {Application, CoreBindings} from '@loopback/core';
+import * as debugFactory from 'debug';
+import {ArtifactOptions} from '../interfaces';
+import {BootBindings} from '../keys';
+import {BaseArtifactBooter} from './base-artifact.booter';
+
+const debug = debugFactory('loopback:boot:interceptor-booter');
+
+type InterceptorProviderClass = Constructor<Provider<Interceptor>>;
+
+/**
+ * A class that extends BaseArtifactBooter to boot the 'InterceptorProvider' artifact type.
+ *
+ * Supported phases: configure, discover, load
+ *
+ * @param app Application instance
+ * @param projectRoot Root of User Project relative to which all paths are resolved
+ * @param [bootConfig] InterceptorProvider Artifact Options Object
+ */
+export class InterceptorProviderBooter extends BaseArtifactBooter {
+  interceptors: InterceptorProviderClass[];
+
+  constructor(
+    @inject(CoreBindings.APPLICATION_INSTANCE)
+    public app: Application,
+    @inject(BootBindings.PROJECT_ROOT) projectRoot: string,
+    @inject(`${BootBindings.BOOT_OPTIONS}#interceptors`)
+    public interceptorConfig: ArtifactOptions = {},
+  ) {
+    super(
+      projectRoot,
+      // Set InterceptorProvider Booter Options if passed in via bootConfig
+      Object.assign({}, InterceptorProviderDefaults, interceptorConfig),
+    );
+  }
+
+  /**
+   * Uses super method to get a list of Artifact classes. Boot each file by
+   * creating a DataSourceConstructor and binding it to the application class.
+   */
+  async load() {
+    await super.load();
+
+    this.interceptors = this.classes as InterceptorProviderClass[];
+    for (const interceptor of this.interceptors) {
+      debug('Bind global interceptor: %s', interceptor.name);
+      const binding = createBindingFromClass(interceptor, {
+        namespace: GLOBAL_INTERCEPTOR_NAMESPACE,
```

Comment:
I am concerned about limiting the files in `src/interceptors` directory to global interceptors. Where are developers supposed to place local (non-global) interceptors?

Personally, I think I would prefer to keep all my interceptors in `src/interceptors` and define a mechanism or convention for deciding which interceptors should be bound as global interceptors; and which should be only bound in the context for manual consumption, or perhaps not bound at all and consumed via `import`.

Two options come to my mind:
1. Use nested directories. `/interceptors/foo.interceptor.ts` is a regular interceptor that is only bound to context but not invoked globally. `/interceptors/global/foo.interceptor.ts` is a global interceptor.
2. Use `@bind`metadata in individual source files to control whether the interceptor should be registered as global or not.

IIUC, you are already leveraging the second approach via `@globalInterceptor('<%= group %>')`, which looks very reasonably to me 👍 

Now let's make sure we can boot non-global interceptors too.

Reply:
Added support for non-global interceptors. I had it in mind but was trying to allow global ones 1st.